doc/transaction_json.rst

```diff
@@ -0,0 +1,139 @@
+..
+  Copyright (C) 2020 Red Hat, Inc.
+
+  This copyrighted material is made available to anyone wishing to use,
+  modify, copy, or redistribute it subject to the terms and conditions of
+  the GNU General Public License v.2, or (at your option) any later version.
+  This program is distributed in the hope that it will be useful, but WITHOUT
+  ANY WARRANTY expressed or implied, including the implied warranties of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+  Public License for more details.  You should have received a copy of the
+  GNU General Public License along with this program; if not, write to the
+  Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+  02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+  source code or documentation are not subject to the GNU General Public
+  License and may only be used or replicated with the express permission of
+  Red Hat, Inc.
+
+.. _transaction_json-label:
+
+################################
+ Stored Transaction JSON Format
+################################
+
+==================
+ Top-level Object
+==================
+
+``version``
+    Type: string
+
+    The version of the stored transaction format, in the form ``MAJOR.MINOR``.
+
+    ``MAJOR`` version denotes backwards incompatible changes (old dnf won't work with
+    new transaction JSON).
+
+    ``MINOR`` version denotes extending the format without breaking backwards
+    compatibility (old dnf can work with new transaction JSON), but breaks forwards
+    compatibility (new dnf won't work with old transaction JSON).
```

Comment:
> new dnf won't work with old transaction JSON

Are you sure about this statement?
New DNF should always work with the old format (unless we explicitly drop the support).

Reply:
Well... let's say you add a field `foo` to the JSON and read it when calling `dnf replay transaction.json`. It will fail on the old files, because `foo` won't be there.

We can of curse handle the case in some way with some kind of a fallback, which we likely will do in most cases, so this statement might be a bit strong... But without a fallback it would be the case.

Should I change the documentation somehow?

Reply:
I am also lost here. Normally I suggest that when I create 2.1 json it will be compatible with 2.2 dnf. The reason is that 2.2 dnf knows changes in 2.1 format, therefore it should be ready for 2.1. format. In other direction 2.1 dnf cannot know what changes are in 2.2 format. Anyway, my expectation is that all version of dnf-2 json format are compatible.